src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -210,13 +211,18 @@ public void run() {
                 connectorConfig.getString(HdfsSinkConnectorConfig.WRITER_LOGGING_SCHEMA_REGISTRY_CONFIG));
 
         writerLogProducer = new KafkaProducer<>(props);
+        writerLogSchema = org.apache.avro.SchemaBuilder.record("writerlog").fields()
```

Comment:
I agree. It is not good to expose the schema to DataWriter. I would propose an alternative solution: add static method in TopicPartitionWriter and call it from DataWriter